homeassistant/__main__.py

```diff
@@ -19,6 +19,48 @@
 from homeassistant.util.async import run_callback_threadsafe
 
 
+import asyncio.tasks
+
+
+def monkey_patch_asyncio():
+    """Replace weakref.WeakSet to address Python 3 bug.
+
+    Under heavy threading operations that schedule calls into
+    the asyncio event loop, Task objects are created. Due to
+    a bug in Python, GC may have an issue when switching between
+    the threads and objects with __del__ (which various components
+    in HASS have).
+
+    This monkey-patch removes the weakref.Weakset, and replaces it
+    with an object that ignores the only call utilizing it (the
+    Task.__init__ which calls _all_tasks.add(self)). It also removes
+    the __del__ which could trigger the future objects __del__ at
+    unpredictable times.
+
+    The side-effect of this manipulation of the Task is that
+    Task.all_tasks() is no longer accurate, and there will be no
+    warning emitted if a Task is GC'd while in use.
+
+    On Python 3.6, after the bug is fixed, this monkey-patch can be
```

Comment:
Does this have any bad side effects if it's still applied to 3.6? As long as 3.4 is supported, this would be needed, and I suspect that would be awhile because 3.4 is the latest on raspbian.

If that's the case, can it dynamically be only applied to python 3.4 and 3.5?